assets/dev/js/admin/gutenberg.js

```diff
@@ -1,7 +1,113 @@
 /* global ElementorGutenbergSettings */
+import GenerateExcerptWithAI from '../../../../modules/ai/assets/js/gutenberg/excerpt';
+import { createRoot } from '@wordpress/element';
+import { RequestIdsProvider } from '../../../../modules/ai/assets/js/editor/context/requests-ids';
+
 ( function( $ ) {
 	'use strict';
 
+	// Wait for the Gutenberg editor to initialize
+	wp.domReady( () => {
+		// Define a function to add the custom link to the excerpt panel
+		const addGenerateExcerptWithAI = () => {
+			// Get the excerpt panel
+			const excerptPanel = document.querySelector( '.editor-post-excerpt' );
+
+			// Check if the excerpt panel exists and the custom link hasn't been added
+			if ( excerptPanel && ! document.querySelector( '.elementor-generate-excerpt-with-ai-link' ) ) {
+				// Create a new link element
+				const generateExcerptWithAILink = document.createElement( 'a' );
+				// Add a class to the link for styling
+				generateExcerptWithAILink.classList.add( 'elementor-generate-excerpt-with-ai-link', 'e-excerpt-ai' );
+				// Add text content to the link
+				generateExcerptWithAILink.textContent = 'Generate with Elementor AI';
+				// Add an event listener to the link
+				generateExcerptWithAILink.addEventListener( 'click', ( event ) => {
+					event.preventDefault();
+
+					const rootElement = document.createElement( 'div' );
+					document.body.append( rootElement );
+
+					const urlSearchParams = new URLSearchParams( window.location.search );
+					elementorCommon.ajax.addRequestConstant( 'editor_post_id', urlSearchParams.get( 'post' ) );
+
+					function onClose() {
+						root.unmount();
+						rootElement.parentNode.removeChild( rootElement );
+					}
+
+					const root = createRoot( rootElement );
+					root.render( <RequestIdsProvider>
```

Comment:
Why not render the HTML elements (`generateExcerptWithAILink`, etc.) as react elements?